### PR TITLE
Fixed #29496 -- Fixed crash on Oracle when converting a non-unique field to primary key.

### PR DESCRIPTION
**Pull request by felixxm**
Ticket [29496](https://code.djangoproject.com/ticket/29496).

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot, test on oracle.

File: django/db/backends/base/schema.py [hunk 1]
```diff
@@ -966,6 +964,14 @@ def _field_indexes_sql(self, model, field):
     def _field_should_be_indexed(self, model, field):
         return field.db_index and not field.unique
 
+    def _primary_key_should_be_added(self, old_field, new_field):
```
Comment (timgraham) [inline django/db/backends/base/schema.py]: I would call this `_field_became_primary_key`.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
@timgraham Thanks for the review.

**Pull request merged**